src/reader/decompress.rs

```diff
@@ -3,13 +3,62 @@
 
 use std::io::Read;
 
-use arrow::datatypes::ToByteSlice;
 use bytes::{Bytes, BytesMut};
 use fallible_streaming_iterator::FallibleStreamingIterator;
 use snafu::ResultExt;
 
 use crate::error::{self, Error};
-use crate::proto::CompressionKind;
+use crate::proto::{self, CompressionKind};
+
+#[derive(Clone, Copy, Debug)]
+pub struct Compression {
+    compression_type: CompressionType,
+    /// No compression chunk will decompress to larger than this size.
+    /// Use to size the scratch buffer appropriately.
+    max_decompressed_block_size: usize,
+}
+
+impl Compression {
+    pub fn from_proto(
+        kind: proto::CompressionKind,
+        compression_block_size: Option<u64>,
+    ) -> Option<Self> {
+        // Spec states default is 256K
+        let max_decompressed_block_size = compression_block_size.unwrap_or(256 * 1024) as usize;
+        match kind {
+            CompressionKind::None => None,
+            CompressionKind::Zlib => Some(Self {
+                compression_type: CompressionType::Zlib,
+                max_decompressed_block_size,
+            }),
+            CompressionKind::Snappy => Some(Self {
+                compression_type: CompressionType::Snappy,
+                max_decompressed_block_size,
+            }),
+            CompressionKind::Lzo => Some(Self {
+                compression_type: CompressionType::Lzo,
+                max_decompressed_block_size,
+            }),
+            CompressionKind::Lz4 => Some(Self {
+                compression_type: CompressionType::Lz4,
+                max_decompressed_block_size,
+            }),
+            CompressionKind::Zstd => Some(Self {
+                compression_type: CompressionType::Zstd,
+                max_decompressed_block_size,
+            }),
+        }
+    }
+}
```

Comment:
Decided to represent no compression as None in Option, since it has different behaviour from when there is compression.

Compression all shares similar behaviour of needing to decode a header for a compressed block whereas no compression doesn't need this, so figured would be more appropriate this way